operator/v1/types_ingress.go

```diff
@@ -1382,6 +1382,97 @@ type IngressControllerHTTPHeaders struct {
+
+// IngressControllerHTTPHeaderActionUnion specifies an action to take on an HTTP header.
+// +union
+type IngressControllerHTTPHeaderActionUnion struct {
+	// type defines the type of the action to be applied on the header.
+	// Possible values are Set and Delete.
+	// Set allows you to set http request and response headers.
+	// Delete allows you to delete http request and response headers
+	// +unionDiscriminator
+	// +kubebuilder:validation:Enum:=Set;Delete
+	// +kubebuilder:validation:Required
+	Type IngressControllerHTTPHeaderActionType `json:"type"`
+
+	// set specifies how the HTTP header should be set.
+	// If this field is empty, no action is taken.
+	// +optional
+	// +unionMember
+	Set *IngressControllerSetHTTPHeader `json:"set,omitempty"`
+}
+
+// IngressControllerHTTPHeaderActionType defines actions that can be performed on HTTP headers.
+type IngressControllerHTTPHeaderActionType string
+
+const (
+	// Set specifies that an HTTP header should be set.
+	Set IngressControllerHTTPHeaderActionType = "Set"
+	// Delete specifies that an HTTP header should be deleted.
+	Delete IngressControllerHTTPHeaderActionType = "Delete"
+)
+
+// IngressControllerSetHTTPHeader defines the value which needs to be set on an HTTP header.
+type IngressControllerSetHTTPHeader struct {
+	// value specifies a header value.
+	// Dynamic values can be added. The value will be interpreted as an HAProxy format string as defined in
+	// http://cbonte.github.io/haproxy-dconv/2.6/configuration.html#8.2.6 and may use HAProxy's %[] syntax and
+	// otherwise must be a valid HTTP header value as defined in https://datatracker.ietf.org/doc/html/rfc7230#section-3.2
+	// +required
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=4096
+	// +kubebuilder:validation:Pattern="^(%(%|(\\{[-+]?[QXE](,[-+]?[QXE])*\\})?([A-Za-z]+|\\[[.0-9A-Z_a-z]+(\\([^)]+\\))?(,[.0-9A-Z_a-z]+(\\([^)]+\\))?)*\\]))|[^%[:cntrl:]])*$"
+	// +kubebuilder:validation:XValidation:rule="!(self.contains('debug(') || self.contains('json_query(') || self.contains('set-var(') || self.contains('map_str(') || self.contains('map_str_int(') || self.contains('map_str_ip(') || self.contains('map_beg(') || self.contains('map_beg_int(') || self.contains('map_end_ip(') || self.contains('map_sub(') || self.contains('map_sub_int(') || self.contains('map_sub_ip(') || self.contains('map_dir(') || self.contains('map_dir_int(') || self.contains('map_dir_ip(') || self.contains('map_dom(') || self.contains('map_dom_int(') || self.contains('map_dom_ip(') || self.contains('map_end(') || self.contains('map_end_int(') || self.contains('map_end_ip(') || self.contains('map_reg(') || self.contains('map_reg_int(') || self.contains('map_reg_ip(') || self.contains('map_regm(') || self.contains('map_int(') || self.contains('map_int_int(') || self.contains('map_int_ip(') || self.contains('map_ip(') || self.contains('map_ip_int(') || self.contains('map_ip_ip('))",message="converter specified not permitted."
```

Comment:
Is the formatting of these single quotes so that the generator generates them correctly? Can we swap them to some other quoting? This looks bizarre at present, but you do seem to have made it work

Reply:
BTW @JoelSpeed, @Miciah has asked me to match against a list of allowed converters. However it seems to be very hard to pick converters from huge list. Also some of them might deprecate in the future. So, I am discussing wheher will it be OK just to check the format of the value rather than checking the allowed list of converters.